File: java/twoSum.java
// Given an array of integers, return indices of the two numbers such that they add up to a specific target.
// https://leetcode.com/problems/two-sum/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i=0;
        HashMap<Integer, Integer> map = new HashMap<>();
        while(!map.containsKey(target - nums[i])){
            map.put(nums[i], i);
            i++;
        }
        return new int[]{map.get(target-nums[i]), i};
    }
}

File: python/twoSum.py
#https://leetcode.com/problems/two-sum/
class Solution(object):
    def twoSum(self, nums, target):
        dict = {}
        for index, num in enumerate(nums):
            if(target - num in dict):
                return [dict.get(target-num), index]
            dict[num] = index